packages/client/Atmosphere.ts

```diff
@@ -314,7 +316,35 @@ export default class Atmosphere extends Environment {
     this.setAuthToken(authToken)
   }
 
-  setAuthToken = (authToken: string | null) => {
+  private validateImpersonation = async (iat: number) => {
+    const isAnotherParabolTabOpen = async () => {
+      const askOtherTabs = new Promise((resolve) => {
+        if (!this.tabCheckChannel) {
+          this.tabCheckChannel = new BroadcastChannel('tabCheck')
```

Comment:
this is the first time we've used a broadcast channel in this app, so it deserves some explanation.

When we open up a new tab, that tab doesn't have an `opener` so we can't see if we opened it or who owns it.
we could do things like attach metadata to that new tab, but that means whenever we open a new tab we'd have to remember to attach that metadata.

introduce broadcast channels-- a way for tabs to talk to one another.
if an impersonate token is detected, it opens a channel.
that means when the e.g. invoice tab gets opened, it pings all the other parabol tabs, sees that 1 is open, so it doesn't log the user out.